### ISSUE TITLE
Make equals/hashcode configurable

### ISSUE THREAD

**Issue by lodart** [javaparser/javaparser#1061]
Hi ! When parsing this portion of code : 
```java
public class Test {
    String a = "google";

    public void test() throws MalformedURLException {
        new URL(a);
        String a = "yahoo";
        new URL(a);
    }
}
```
I realised both `NameExpr` `a` share the same HashCode, so I can't use them as a key in a HashMap. `.equals` returns also true. Is this normal behaviour ?

Thank you :)

**Comment by matozoid** [comment]
I wanted to say "yes, that is expected behaviour, but you can change it by changing the visitors used by `Node`" - but it looks like that wasn't implemented yet. Putting it on the backlog for next release!

**Comment by lodart** [comment]
Thank you :)

**Comment by matozoid** [comment]
Oh, I've been thinking this over for a bit: when I do what I originally planned, things will break horribly. A lot of code depends on the default behaviour of equals and hashcode in `Node`. So I'm going to see if I can make a `VisitorMap` :-)

**Comment by matozoid** [comment]
@lodart would it be possible for you to take the source code from this repo: https://github.com/matozoid/javaparser.git

branch `visitor_map`

and see if this works for you?

**Comment by lodart** [comment]
Unfortunately, I don't think it has worked :/ [Here's what I got](https://lut.im/vSlEhtcpgw/gCdfvZLDoz71sqph). I must have done something wrong, though. I struggled adding the library to IntelliJ (I'm used to just copying Maven dependencies :) ).
So I cloned the repo (the branch `visitor_map`), I did `mvn clean install`, then I imported the module and added the module dependency. Then, [I removed some stuff that didn't compile](https://lut.im/IbPBA8iWw3/Hcx9zZjmzVYh9tFd) and here I am ...
